tests/api.c

```diff
@@ -38825,6 +38828,21 @@ static void test_wolfSSL_CTX_ctrl(void)
     AssertIntEQ(wolfSSL_EC_KEY_generate_key(ecKey),1);
 #endif
 
+#ifdef HAVE_ED25519
+    {
+        WOLFSSL_EVP_KEY* pKey;
```

Comment:
You didn't test run this section at all. Please make sure that what you submit for review compiles.
```
julek@julek-MS-7A38:~/Projects/wolfssl-julek$ make
make -j13  all-recursive
make[1]: Entering directory '/home/julek/Projects/wolfssl-julek'
make[2]: Entering directory '/home/julek/Projects/wolfssl-julek'
make[2]: warning: -j13 forced in submake: resetting jobserver mode.
  CC       wolfcrypt/benchmark/benchmark.o
  CC       wolfcrypt/src/src_libwolfssl_la-hmac.lo
  CC       wolfcrypt/src/src_libwolfssl_la-sha256.lo
  CC       wolfcrypt/src/src_libwolfssl_la-hash.lo
  CC       wolfcrypt/src/src_libwolfssl_la-cpuid.lo
  CC       wolfcrypt/src/src_libwolfssl_la-random.lo
  CC       wolfcrypt/src/src_libwolfssl_la-sha512.lo
  CC       wolfcrypt/src/src_libwolfssl_la-aes.lo
  CC       wolfcrypt/src/src_libwolfssl_la-rsa.lo
  CC       wolfcrypt/src/src_libwolfssl_la-sha.lo
  CC       wolfcrypt/src/src_libwolfssl_la-sha3.lo
  CC       wolfcrypt/src/src_libwolfssl_la-logging.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_port.lo
  CC       wolfcrypt/src/src_libwolfssl_la-error.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_encrypt.lo
  CC       wolfcrypt/src/src_libwolfssl_la-signature.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wolfmath.lo
  CC       wolfcrypt/src/src_libwolfssl_la-memory.lo
  CC       wolfcrypt/src/src_libwolfssl_la-dh.lo
  CC       wolfcrypt/src/src_libwolfssl_la-asn.lo
  CC       wolfcrypt/src/src_libwolfssl_la-coding.lo
  CC       wolfcrypt/src/src_libwolfssl_la-poly1305.lo
  CC       wolfcrypt/src/src_libwolfssl_la-md5.lo
  CC       wolfcrypt/src/src_libwolfssl_la-pwdbased.lo
  CC       wolfcrypt/src/src_libwolfssl_la-pkcs12.lo
  CC       wolfcrypt/src/src_libwolfssl_la-chacha.lo
  CC       wolfcrypt/src/src_libwolfssl_la-chacha20_poly1305.lo
  CC       wolfcrypt/src/src_libwolfssl_la-tfm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ecc.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ed25519.lo
  CC       wolfcrypt/src/src_libwolfssl_la-fe_operations.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ge_operations.lo
  CC       src/libwolfssl_la-internal.lo
  CC       src/libwolfssl_la-wolfio.lo
  CC       src/libwolfssl_la-keys.lo
  CC       src/libwolfssl_la-ssl.lo
  CC       src/libwolfssl_la-tls.lo
  CC       src/libwolfssl_la-tls13.lo
  CC       src/libwolfssl_la-ocsp.lo
  CC       wolfcrypt/test/test.o
  CC       examples/benchmark/tls_bench.o
  CC       examples/client/client-client.o
  CC       examples/echoclient/echoclient.o
  CC       examples/echoserver/echoserver.o
  CC       examples/server/server-server.o
  CC       wolfcrypt/test/testsuite_testsuite_test-test.o
  CC       examples/client/testsuite_testsuite_test-client.o
  CC       examples/echoclient/testsuite_testsuite_test-echoclient.o
  CC       examples/echoserver/testsuite_testsuite_test-echoserver.o
  CC       examples/server/testsuite_testsuite_test-server.o
  CC       testsuite/testsuite_test-testsuite.o
  CC       tests/unit_test-unit.o
  CC       tests/unit_test-api.o
  CC       tests/unit_test-suites.o
  CC       tests/unit_test-hash.o
  CC       tests/unit_test-srp.o
  CC       examples/client/tests_unit_test-client.o
  CC       examples/server/tests_unit_test-server.o
tests/api.c: In function ‘test_wolfSSL_CTX_ctrl’:
tests/api.c:38833:9: error: unknown type name ‘WOLFSSL_EVP_KEY’; did you mean ‘WOLFSSL_EVP_PKEY’?
38833 |         WOLFSSL_EVP_KEY* pKey;
      |         ^~~~~~~~~~~~~~~
      |         WOLFSSL_EVP_PKEY
In file included from tests/api.c:134:
tests/api.c:38838:23: error: ‘pkey’ undeclared (first use in this function); did you mean ‘pKey’?
38838 |         AssertNotNull(pkey = X509_get_pubkey(X509));
      |                       ^~~~
./tests/unit.h:50:49: note: in definition of macro ‘Assert’
   50 | #define Assert(test, description, result) if (!(test)) Fail(description, result)
      |                                                 ^~~~
tests/api.c:38838:9: note: in expansion of macro ‘AssertNotNull’
38838 |         AssertNotNull(pkey = X509_get_pubkey(X509));
      |         ^~~~~~~~~~~~~
tests/api.c:38838:23: note: each undeclared identifier is reported only once for each function it appears in
38838 |         AssertNotNull(pkey = X509_get_pubkey(X509));
      |                       ^~~~
./tests/unit.h:50:49: note: in definition of macro ‘Assert’
   50 | #define Assert(test, description, result) if (!(test)) Fail(description, result)
      |                                                 ^~~~
tests/api.c:38838:9: note: in expansion of macro ‘AssertNotNull’
38838 |         AssertNotNull(pkey = X509_get_pubkey(X509));
      |         ^~~~~~~~~~~~~
tests/api.c:38838:46: error: expected expression before ‘X509’
38838 |         AssertNotNull(pkey = X509_get_pubkey(X509));
      |                                              ^~~~
./tests/unit.h:50:49: note: in definition of macro ‘Assert’
   50 | #define Assert(test, description, result) if (!(test)) Fail(description, result)
      |                                                 ^~~~
tests/api.c:38838:9: note: in expansion of macro ‘AssertNotNull’
38838 |         AssertNotNull(pkey = X509_get_pubkey(X509));
      |         ^~~~~~~~~~~~~
tests/api.c:38833:26: error: unused variable ‘pKey’ [-Werror=unused-variable]
38833 |         WOLFSSL_EVP_KEY* pKey;
      |                          ^~~~
tests/api.c:38851:19: error: declaration of ‘pkey’ shadows previous non-variable [-Werror=shadow]
38851 |         EVP_PKEY* pkey;
      |                   ^~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:5647: tests/unit_test-api.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/julek/Projects/wolfssl-julek'
make[1]: *** [Makefile:5982: all-recursive] Error 1
make[1]: Leaving directory '/home/julek/Projects/wolfssl-julek'
make: *** [Makefile:3612: all] Error 2
```